src/data/upgrade.ts

```diff
@@ -0,0 +1,19 @@
+import { Event, UpgradeData } from "@clarity-types/data";
+import config from "@src/core/config";
+import encode from "@src/data/encode";
+
+export let data: UpgradeData = null;
+
+export function reset(): void {
+    data = null;
+}
+
+// Following call will upgrade the session from lean mode into the full mode retroactively from the start of the page.
```

Comment:
Who calls this function and when? It is supposed to be externally triggered?

Reply:
Yes, you will notice the call in SNR code flow that I will send out shortly. Similar to start function too - it's called externally. 

---
In reply to: [369061624](https://github.com/microsoft/clarity-js/pull/166#discussion_r369061624) [](ancestors = 369061624)